### PR TITLE
Add validation to not admit RoleTemplates with both context=cluster & ProjectCreatorDefault=true

### PR DESCRIPTION
**Pull request by vatsalparekh**
## Issue: 

Add validation to not admit RoleTemplates with both context=cluster & ProjectCreatorDefault=true

## Problem

SURE-7635

## Solution
Add a check for this field in validation

## CheckList
  
- [x] Test
  
- [x] Docs

### CODE REVIEW

**Comment by samjustus** [comment]
https://github.com/rancher/rancher/issues/44860

File: tests/integration/roleTemplate_test.go [hunk 1]
```diff
@@ -118,3 +118,35 @@ func (m *IntegrationSuite) TestRoleTemplateNoAPIGroups() {
 	}
 	validateEndpoints(m.T(), endPoints, m.clientFactory)
 }
+
+func (m *IntegrationSuite) TestRoleTemplateInvalidContext() {
+	newObj := func() *v3.RoleTemplate { return &v3.RoleTemplate{} }
```
Comment (bigkevmcd) [inline tests/integration/roleTemplate_test.go:123]: This test can maybe be named a bit better?

Someone reading it would need to figure out that the only significant thing in this test is the `invalidCreate.ProjectCreatorDefault` line?

Also, maybe we could improve the validation a bit?

**Review by bigkevmcd — COMMENTED** [review]
While I think the actual fix is fine, I think we can improve the tests to clarify that they're testing.

File: tests/integration/roleTemplate_test.go (hunk 1 above)
Comment (vatsalparekh) [reply tests/integration/roleTemplate_test.go:123]: > maybe we could improve the validation a bit?

can you elaborate how

File: pkg/resources/management.cattle.io/v3/roletemplate/validator.go [hunk 2]
```diff
@@ -192,6 +192,9 @@ func validateCreateFields(newRole *v3.RoleTemplate, fldPath *field.Path) *field.
 }
 
 func validateContextValue(newRole *v3.RoleTemplate, fldPath *field.Path) *field.Error {
+	if newRole.Context == clusterContext && newRole.ProjectCreatorDefault == true {
```
Comment (JonCrowther) [inline pkg/resources/management.cattle.io/v3/roletemplate/validator.go:195]: In the docs you stated

> If the `roletemplate.ProjectCreatorDefault` is true, context must equal `"project"`

But this checks only that the context is `"cluster"`. Context can be `"cluster"` `"project"` or `""` ([source](https://github.com/rancher/rancher/blob/c5b4413739fb7a9f935eadeb7723b28b90ee22be/pkg/apis/management.cattle.io/v3/authz_types.go#L280)), so this check should probably be:
```suggestion
	if newRole.Context != projectContext && newRole.ProjectCreatorDefault == true {
```

Alternatively, the doc should say 

> If the `roletemplate.ProjectCreatorDefault` is true, context can not be `"cluster"`

File: tests/integration/roleTemplate_test.go [hunk 3]
```diff
@@ -118,3 +121,32 @@ func (m *IntegrationSuite) TestRoleTemplateNoAPIGroups() {
...
+	m.Require().NoError(err, "Failed to get gvk")
+	client, err := m.clientFactory.ForKind(gvk)
+	m.Require().NoError(err, "Failed to create client")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	err = client.Create(ctx, invalidRoleTemplate.GetNamespace(), invalidRoleTemplate, nil, v1.CreateOptions{})
+	m.Assert().NotNil(err)
+	m.Assert().Contains(err.Error(), "RoleTemplate context must be Project when projectCreatorDefault=true")
```
Comment (bigkevmcd) [inline tests/integration/roleTemplate_test.go:151]: I don't think this error is right?
```suggestion
	m.Assert().Contains(err.Error(), "RoleTemplate context must be project when projectCreatorDefault=true")
```

File: tests/integration/roleTemplate_test.go [hunk 4]
```diff
@@ -118,3 +121,32 @@ func (m *IntegrationSuite) TestRoleTemplateNoAPIGroups() {
...
+	gvk, err := m.clientFactory.GVKForObject(invalidRoleTemplate)
+	m.Require().NoError(err, "Failed to get gvk")
+	client, err := m.clientFactory.ForKind(gvk)
+	m.Require().NoError(err, "Failed to create client")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	err = client.Create(ctx, invalidRoleTemplate.GetNamespace(), invalidRoleTemplate, nil, v1.CreateOptions{})
+	m.Assert().NotNil(err)
```
Comment (bigkevmcd) [inline tests/integration/roleTemplate_test.go:150]: You don't need this check, because the test will fail if it's not `nil`.

File: tests/integration/roleTemplate_test.go (hunk 3 above)
Comment (vatsalparekh) [reply tests/integration/roleTemplate_test.go]: Fixed

**Review by JonCrowther — APPROVED** [review]

**Review by bigkevmcd — APPROVED** [review]
Looks good, thank you!
